### ISSUE TITLE
Data received > 100%?

### ISSUE THREAD

**Issue by Sbte** [emesene/emesene#277]
I tried to make something reasonable out of the elapsed time after you sent or received the file. What I have works for received files, but for some reason when you send a file the received_data goes over the size. FileTransfer.RECEIVED is also never set (AFAIK), because finished() in FileTransferWidget is never called. 

When I look at the tooltip, it says for example 139% (42/29 KB). It comes with this error:

/home/sven/emesene-dev/emesene/gui/gtkui/FileTransferWidget.py:104: GtkWarning: gtk_progress_set_percentage: assertion `percentage >= 0 && percentage <= 1.0' failed

logically,

if self.transfer.state == self.transfer.RECEIVED:
            self.progress.set_fraction(1)  # 100%

doesn't work, because transfer.RECEIVED is never set.

And I get a lot of those or similar messages when sending or receiving the file:

[16:52:05 WARNING papyon.msnp2p.transport.switchboard] Invalid TLP chunk in SB message: TLPv1 Parse Error: empty chunk for non-empty blob
a8 c1 49 0b 84 10 ad 3d 00 00 00 00 00 00 00 00
04 00 00 00 00 00 00 00 00 00 00 00 02 00 00 00
c2 dc f1 76 8b e2 d5 1c 04 00 00 00 00 00 00 00

**Comment by zipatei** [comment]
one solutions for this:

In e3 base transfer change this lines:

```
def get_fraction(self):
    ''' received the percentage, which is < 1 and > 0 '''
    return (float(self.received_data) / self.size)
```

For this:

```
def get_fraction(self):
    ''' received the percentage, which is < 1 and > 0 '''
    if self.received_data <= self.size:
        return (float(self.received_data) / self.size)
    else:
        return (float(1))
```

**Comment by zipatei** [comment]
About the error:

[13:14:43 WARNING papyon.msnp2p.transport.switchboard] Invalid TLP chunk in SB message: TLPv1 Parse Error: empty chunk for non-empty blob
92 90 1e 28 62 b4 4a 26 00 00 00 00 00 00 00 00
04 00 00 00 00 00 00 00 00 00 00 00 02 00 00 00
85 f3 32 29 a3 6f f0 30 04 00 00 00 00 00 00 00
/home/andre/emesene/emesene/gui/gtkui/FileTransferWidget.py:102: GtkWarning: IA__gtk_progress_set_percentage: assertion `percentage >= 0 && percentage <= 1.0' failed
  self.progress.set_fraction(self.transfer.get_fraction())

And it seems to only occur when offline.

**Comment by c10ud** [comment]
the filetransfer stuff is a quick and dirty port from emesene-1 so yes, it needs some love in order to work correctly.. if you've found a solution please tell me or send a pull req :D

**Comment by Sbte** [comment]
zipatei:

def get_fraction(self):
    ''' received the percentage, which is < 1 and > 0 '''
    if self.received_data <= self.size:
        return (float(self.received_data) / self.size)
    else:
        return (float(1))

This only fixes the percentage, not the fact that the received_data is bigger than the size, and thus the RECEIVED state is never reached and things are messed up. 

Tested this with sending to emesene 1 and emesene 2, both with emesene2, but the both appear broken in emesene 2 (only when sending). In emesene 1 the sending works fine.

**Comment by c10ud** [comment]
Filetransfers small update.
Closed by 103a33c3087b1d98dbacd491de886ea1e577c898
Closed by 103a33c3087b1d98dbacd491de886ea1e577c898
Closed by 103a33c3087b1d98dbacd491de886ea1e577c898

**Comment by Sbte** [comment]
This wasn't fixed. Could you reopen this?
